docs/website/contents/for-developers/QueryVersioning.md

```diff
@@ -39,13 +44,28 @@ Our code does not use the negotiated [`NodeToClientVersion`][n2c] directly, but
 
 Additionally, all versions larger than a certain `NodeToClientVersion` (see [`latestReleasedNodeVersion`][latestReleasedNodeVersion]) are considered experimental, and are only offered in the initial version negotiation when a flag (currently, `ExperimentalProtocolsEnabled`) is set. Also see [`limitToLatestReleasedVersion`][limitToLatestReleasedVersion] and its call/usage sites.
 
+#### Shelly node-to-client version
+
+Each `ShelleyNodeToClientVersion` has a set of queries it supports. Assume the maximum version is $X$, and that it has queries $Q_0$, ... $Q_{n-1}$ associated to it. If no node was released that supports version $X$, ie `ShelleyNodeToClientVersionX`, we have a reasonable degree of certainty that no node will send any $Q_i$, $x \in [0, n)$ to older nodes (since no such node was yet released). Therefore, if we add a new query $Q_n$ we can associate it to the unreleased version $X$ (`ShelleyNodeToClientVersionX`).
+
+On the other hand, the node that supports version `X` has been released, then we
+need to increase the maximum Shelley node-to-client version, by adding one more constructor to `ShelleyNodeToClientVersion`, which is defined in module [Ouroboros.Consensus.Shelley.Ledger.NetworkProtocolVersion](https://github.com/input-output-hk/ouroboros-consensus/blob/main/ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/NetworkProtocolVersion.hs). By adding this new version the node is able to detect if other Cardano clients that respect this versioning mechanism support said query.
+
+Henceforth, we call an unreleased version "experimental" (ie only used for demo purposes/specific to an unreleased era).
+
 ### Checks
 
 The supported query versions are only enforced in the [Shelley query encoder][shelley-encodeNodeToClient], ie in code run by clients. The server will answer queries even if the negotiated version is older than expected (which might be the case with a custom client implementation), following the robustness principle.
 
+### On newly added golden files
+
+When adding a new `ShelleyNodeToClientVersion` new golden files will be generated. Because serialization is [version dependent](https://github.com/input-output-hk/ouroboros-consensus/pull/95), a new `ShelleyNodeToClientVersion` could also introduce a different serialization. See how function [`decodeShelleyResult`](https://github.com/input-output-hk/ouroboros-consensus/pull/95/files#diff-0bff1ce7a3c11a5b7371e38d073430fc4e49042adad2ce81577793d0e5d28020R726) uses `ShelleyNodeToClientVersion`. Therefore we have to test the serialization for the new versions.
```

Comment:
This also happens for new `CardanoNodeToClientVersion`s (in `cardano-test`).

Reply:
TODO: include the clarification above.